histogram/histogram_test.go

```diff
@@ -0,0 +1,109 @@
+package histogram
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func Example() {
```

Comment:
What is this?

Reply:
oh dude, package level godoc examples are great (see https://golang.org/pkg/testing/#hdr-Examples) - they get compiled and run as part of tests, and put some nice stuff into the godoc itself